src/cacheExchange.ts

```diff
@@ -250,6 +252,8 @@ export const cacheExchange = (opts?: CacheExchangeOpts): Exchange => ({
       } else {
         data = query(store, operation, data).data;
       }
+    } else if (hasOptimistic) {
+      writeDependencies = store.data.optimisticDependencies[key];
```

Comment:
This seems rather inelegant. Dependencies have nothing to do with `store.data`, so they shouldn't be stored there. Let's swap out `optimisticKeys` being a `Set` for the a `Map` that's similar to `DependentOperations` or being the same even.